tests/views/test_registry_instrument_view.py

```diff
@@ -0,0 +1,206 @@
+import base64
+import json
+from pathlib import Path
+from unittest.mock import patch
+
+from flask import current_app
+
+from tests.test_client import TestClient
+
+api_root = "/collection-instrument-api/1.0.2"
+exercise_id = "3ff59b73-7f15-406f-9e4d-7f00b41e85ce"
+form_type_exists = "0001"
+form_type_does_not_exist = "9999"
+
+
+class TestRegistryInstrumentView(TestClient):
+    """
+    Registry Instrument view unit tests
+    """
+
+    def test_get_registry_instrument_returns_200_and_object(self):
+        with patch(
+            "application.views.registry_instrument_view."
+            "RegistryInstrument.get_registry_instrument_by_exercise_id_and_formtype"
+        ) as mock_get_registry_instrument_by_exercise_id_and_formtype:
+            with open(Path(__file__).parent.parent / "test_data" / "registry_instrument.json") as f:
+                mock_get_registry_instrument_by_exercise_id_and_formtype.return_value = json.load(f)
+            response = self.client.get(
+                f"{api_root}/registry-instrument/exercise-id/{exercise_id}/formtype/{form_type_exists}",
+                headers=self.get_auth_headers(),
+            )
+            self.assertStatus(response, 200)
+            self.assertEqual(len(response.json), 8)  # The 8 fields of a registry instrument
+            self.assertEqual(response.headers["Content-Type"], "application/json")
+            mock_get_registry_instrument_by_exercise_id_and_formtype.assert_called_once_with(
+                exercise_id, form_type_exists
+            )
+
+    def test_get_registry_instrument_returns_404(self):
+        with patch(
+            "application.views.registry_instrument_view."
+            "RegistryInstrument.get_registry_instrument_by_exercise_id_and_formtype"
+        ) as mock_get_registry_instrument_by_exercise_id_and_formtype:
+            mock_get_registry_instrument_by_exercise_id_and_formtype.return_value = None
+            response = self.client.get(
+                f"{api_root}/registry-instrument/exercise-id/{exercise_id}/formtype/{form_type_does_not_exist}",
+                headers=self.get_auth_headers(),
+            )
+            self.assertStatus(response, 404)
+            self.assertEqual(response.data.decode(), "Not Found")  # The 8 fields of a registry instrument
```

Comment:
Not sure on the comment? Feels like a copy paste error

Reply:
Done